src/NHibernate/Id/Enhanced/IAccessCallback.cs

```diff
@@ -10,5 +10,10 @@ public partial interface IAccessCallback
 		/// Retrieve the next value from the underlying source.
 		/// </summary>
 		long GetNextValue();
+
+		/// <summary>
+		/// Obtain the tenant identifier (multi-tenancy), if one, associated with this callback.
+		/// </summary>
+		string GetTenantIdentifier();
```

Comment:
Do we need this?

Reply:
I would say so, yes. Somehow the tenant identifer must become available to the optimizer. This is the way I cloned from https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/id/enhanced/AccessCallback.java , and I don't think there's a better way.